R/translate-sql-base.r

```diff
@@ -273,3 +273,49 @@ base_no_win <- sql_translator(
   lag          = win_absent("lag"),
   order_by     = win_absent("order_by")
 )
+
+#' @export
+#' @rdname sql_variant
+#' @format NULL
+base_odbc_scalar <- sql_translator(
```

Comment:
I think this should live in `db-odbc.R`, and then we should probably call the other files `db-odbc-msql.R`, `db-odbc-hive.R` etc